Manatee.Json/Schema/SchemaValidationContext.cs

```diff
@@ -113,10 +113,12 @@ public SchemaValidationContext(SchemaValidationContext source)
 			Root = source.Root;
 			RecursiveAnchor = source.RecursiveAnchor;
 			Instance = source.Instance;
-			EvaluatedPropertyNames.UnionWith(source.EvaluatedPropertyNames);
-			LocallyEvaluatedPropertyNames.UnionWith(source.LocallyEvaluatedPropertyNames);
-			ValidatedIndices.UnionWith(source.ValidatedIndices);
-			LocallyValidatedIndices.UnionWith(source.LocallyValidatedIndices);
+
+			_evaluatedPropertyNames = new HashSet<string>(source.EvaluatedPropertyNames.AsEnumerable<string>());
```

Comment:
The `.Union()` avoids having to create a new collection instance.  Allocations are slow, so we don't want to do it.

More importantly here, your change doesn't consider the values that already may exist in the collection.  You're just overwriting it.

Reply:
I must be missing something, but I'm ok with that :) As I read the code it's in a public constructor that calls an internal constructor. The internal constructor that is called doesn't touch those HashSets so they should be empty/null in the public constructor. No one should have had the time to call the properties that initialize the HashSets yet since we're still in the constructor so it's not an extra new either.

Reply:
Ah, my bad.  Didn't look at the context of where your change was.  Thought it was in the update method below.

Reply:
I can't comment on the right lines because they're not part of the change, but the fields should no longer be typed as nullable, and the properties need to have the lazy initialization removed.

I think you'll be good then.

Reply:
I think the lazy loading still has to be there because there is code in JsonSchema.cs that uses the internal constructor. I could also add a check in the public contructor that the fields haven't been initialized by another constructor. That would prevent a potential future bug becaus of changes in other constructors.

Reply:
Ah... yes.  That's why I had initialized those that way.  The system is recursive, so the context information is passed from one level to the next through the internal constructor.

I recently made some updates to the serializer that preserved the context and all of its components.  It sped up things quite a bit.  I'd like to eventually do the same for the schema context, but haven't gotten around to it.